feat: Add View recordings to Web analytics

veryayskiy opened the pull request:
We want more people to understand how important to watch recordings.
For that we are looking into other places within Posthog where people can find out about the feature.
One of them is web analytics.
We added a column "Recordings" to the tables to redirect users to "Session Replay" with preset filters.

Before:
<img width="1236" alt="Screenshot 2025-01-21 at 13 28 25" src="https://github.com/user-attachments/assets/5ae0aa2e-b8d5-43d5-a3f7-1f8af64c01cc" />


After:
<img width="1221" alt="Screenshot 2025-01-21 at 21 00 09" src="https://github.com/user-attachments/assets/e5e9db14-8f62-4529-b95a-4321c9e0ec32" />

veryayskiy committed:
feat: Add View recordings to Web analytics

veryayskiy committed:
feat: Add View recordings to Web analytics

RenderReplayButton

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics

joshsny commented:
This is a great one!

I've just added in product cross-sell tracking which the growth team will keep a track of, would you mind adding a `trackProductCrossSell` call to these, along with a `ProductCrossSellLocation`?

You can see an example [here](https://github.com/PostHog/posthog/pull/27483/files) (you might need to merge in master to update)

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics

veryayskiy committed:
Update WebAnalyticsTile.tsx

[+] add handle is the value is empty
[*] a bit of refactoring
[+] separate case for UTM s/m/c
[*] mapping instead of switch

veryayskiy committed:
Merge branch 'feat/sr/web-analytics' of https://github.com/PostHog/posthog into feat/sr/web-analytics

veryayskiy committed:
Fix python issues

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics

veryayskiy committed:
Update WebAnalyticsTile.tsx

[+] add cross-sell

veryayskiy commented on frontend/src/scenes/web-analytics/tiles/WebAnalyticsTile.tsx:
```diff
@@ -661,3 +682,203 @@ export const WebQuery = ({
+        targetBlank: true,
+        onClick: (e: React.MouseEvent) => {
+            e.stopPropagation()
+            trackProductCrossSell({
```
@joshsny added. correct?

joshsny replied:
This is great - just need to add a new location e.g ProductCrossSellLocation.WEB_ANALYTICS_INSIGHT as we track by that also, thanks!

veryayskiy replied:
@joshsny done

veryayskiy committed:
Update WebAnalyticsTile.tsx

[*] fix pathname

veryayskiy committed:
Update WebAnalyticsTile.tsx

[*] made buttons looks nicer

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

veryayskiy committed:
Fix universalFilterbutton

veryayskiy committed:
Merge branch 'feat/sr/web-analytics' of https://github.com/PostHog/posthog into feat/sr/web-analytics

veryayskiy committed:
[*] update cross sell location

[*] update cross sell location

veryayskiy committed:
[*] change UI to make it nicer

joshsny commented on frontend/src/scenes/web-analytics/tiles/WebAnalyticsTile.tsx:
```diff
@@ -661,3 +682,194 @@ export const WebQuery = ({
+                                    {
+                                        key: '$entry_utm_campaign',
+                                        type: PropertyFilterType.Session,
+                                        value: [values[2]],
```
note: `values[2]` feels potentially dangerous here without a length check?

joshsny commented on frontend/src/scenes/web-analytics/tiles/WebAnalyticsTile.tsx:
```diff
@@ -661,3 +682,194 @@ export const WebQuery = ({
+    const type = typeMap[breakdownBy] || PropertyFilterType.Person
+
+    /** Map breakdown types to their corresponding property filter key */
+    const breakdownKeyMap: Partial<Record<WebStatsBreakdown, string>> = {
```
nit(non-blocking): might be worth pulling these (and some of the functions) out of the render to make it easier to read?

joshsny reviewed (commented):
LGTM, will be a great link up between products! 😄

Will leave someone from web analytics to approve since lacking the relative context for ✅

veryayskiy committed:
Optimisation

veryayskiy committed:
merge with master

veryayskiy committed:
make ruff format happy

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics

pauldambra commented on frontend/src/scenes/web-analytics/tiles/WebAnalyticsTile.tsx:
```diff
@@ -367,6 +376,18 @@ export const webAnalyticsDataTableQueryContext: QueryContext = {
         action_name: {
             title: 'Action',
         },
+        replay_url: {
```
do we want to conditionally add the column so we can A/B test it? 
or just add it and track user behavior with insights/replay?

(open question, no push in either direction from me)

veryayskiy committed:
Check dateRange

pauldambra reviewed (approved):
interaction is perfect iMHO 

will let @robbie-c or @rafaeelaudibert review too

robbie-c commented on posthog/hogql_queries/web_analytics/stats_table.py:
```diff
@@ -455,6 +455,26 @@ def calculate(self):
             )
 
+        # Add replay URL column if it doesn't exist
+        if columns is not None and "context.columns.replay_url" not in columns:
```
Looks great, needs a test :D

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

veryayskiy committed:
more edge cases

veryayskiy committed:
merge with master

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics

veryayskiy committed:
update the tests

pauldambra committed:
Merge branch 'master' into feat/sr/web-analytics

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

veryayskiy committed:
try another approach

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics

pauldambra committed:
nicer test failure messages

veryayskiy committed:
update tests

veryayskiy committed:
update tests

veryayskiy committed:
update tests

veryayskiy committed:
Merge branch 'master' into feat/sr/web-analytics
